framework/src/utils/PetscSupport.C

```diff
@@ -733,7 +733,17 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
     // Do not add duplicate settings
     if (find(po.inames.begin(), po.inames.end(), petsc_options_inames[i]) == po.inames.end())
     {
+#if !PETSC_VERSION_LESS_THAN(3, 9, 0)
+      if (petsc_options_inames[i] == "-pc_factor_mat_solver_package")
+      {
+        po.inames.push_back("-pc_factor_mat_solver_type");
```

Comment:
Why not just replace the option instead of adding the new one to the end? You are adding to the vector while you are iterating over it, which probably isn't the best idea. Since you are using indices, it's not wrong but not the most elegant solution. How about:
```
petsc_option_inames[i] = "-pc_factor_mat_solver_type"
```

Reply:
po is the actual petsc options we are going to add to PETSc, and it is filled in this for loop.  petsc_option_inames is something from parser, and we do not need to change it at all.

Reply:
👍